### PR TITLE
Replace Twitter with X

### PR DESCRIPTION
**Pull request by star1327p**
Please replace Twitter with X.

This is related to the PR #24320.

### CODE REVIEW

File: doc/neps/nep-0046-sponsorship-guidelines.rst [hunk 1]
```diff
@@ -86,7 +86,7 @@ Sponsors will get acknowledged through:
 - prominent logo placement on https://numpy.org/about/
 - logos displayed in talks about NumPy by maintainers
 - announcements of the sponsorship on the NumPy mailing list and the numpy-team
-  Twitter account
+  X account on https://x.com/numpy_team/
```
Comment (mattip) [inline doc/neps/nep-0046-sponsorship-guidelines.rst:89]: I think I would prefer to not commit to publishing on any social media, i.e.
```
- announcements of the sponsorship on the NumPy mailing list.
```

**Review by InessaPawson — DISMISSED** [review]

File: doc/neps/nep-0046-sponsorship-guidelines.rst (hunk 1 above)
Comment (InessaPawson) [reply doc/neps/nep-0046-sponsorship-guidelines.rst:89]: @star1327p Please delete this bullet point to finalize the PR.

Comment (star1327p) [reply doc/neps/nep-0046-sponsorship-guidelines.rst]: @InessaPawson Just made the change. Thanks for reviewing!

**Comment by mattip** [comment]
Thanks @star1327p

**Pull request merged**